### DISCUSSION TITLE
The concept of namespace is not easy to understand?

### DISCUSSION THREAD

**Discussion by lepdou** [apolloconfig/apollo#4098] [General]
### What is Namespace？
Namespace in Apollo originally means an abstract configuration set, which is used to realize separate configuration, permission and other management. Concretely speaking, it corresponds to the traditional "configuration file".

### What is problem？
Many users do not understand the concept of namespace. The reasons are as follows:

1. In recent years k8s and many components have introduced the concept of namespace, so when users understand Apollo's namespace, they will naturally refer to the meaning of other component namespaces. However, due to the different meaning of Apollo's namespace and other components, there are differences in understanding.

2. Namespace itself is very abstract, which makes it more difficult for users to understand.

### How to do?
Can you take the opportunity of version 2.0 to replace the namespace with other better understood terms, such as configuration file (配置文件).

### What points to change?

1. Modify the document displayed to users on portal. For example,"create a configuration file" instead of “create a namespace”. A prompt is added on the page to prompt the user to change the namespace concept into a configuration file

2. The API and configuration items in the Apollo client, such as < Apollo:config namespaces = "FX. Apollo, application" / > are changed to < Apollo:config config-files = "FX. Apollo, application" / >. Of course, it needs to be compatible with the old way.

The core API of Apollo client has used a lot of concepts of config and configFile, such as ConfigService.getConfigFile(). So it seems more reasonable.

3. Do not change the core interface and model of the server

### What do you think?

Welcome to leave a message and put forward your ideas~

**Comment by nobodyiam** [comment]
> "create a namespace" instead of "create a configuration file"

I suppose what you mean is "create a configuration file" instead of "create a namespace"?

I agree that the namespace concept is now not easy to understand by normal users, as it's widely used for other scenarios like k8s.
The namespace concept was originally designed as an abstract concept of a collection of configurations. Concretely speaking, it corresponds to the traditional configuration files. So I think we may try to rename it or add tips in places like UI and SDKs.

**Reply by lepdou** [reply]
> I suppose what you mean is "create a configuration file" instead of "create a namespace"?

Yes, It's a mistake.

**Reply by lepdou** [reply]
I will try to do it.
